src/frequenz/channels/_anycast.py

```diff
@@ -158,14 +159,18 @@ class Receiver(BaseReceiver[T]):
     method.
     """
 
+    @dataclass
+    class _Value(Generic[U]):
+        value: U
```

Comment:
Maybe add a comment about why this wrapper is needed? I guess it is because `_next` being `None` has a special meaning, right? Another option would be to have a flag `_has_next`, but I'm good with this if the reason is more obvious.

Reply:
This could be done with a sentinel,  but I was too lazy to look up the syntax.  Let me do that now.

Reply:
I've added a fixup commit that switches to use of a sentinel.